src/Tokenizer/Token.php

```diff
@@ -20,12 +20,21 @@
  * Representation of single token.
  * As a token prototype you should understand a single element generated by token_get_all.
  *
+ * @phpstan-type _TokenKind int|string
+ *
  * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
  *
  * @readonly
  */
 final class Token
 {
+    /**
+     * Kind of token.
```

Comment:
> Term | `new Token([T_CLASS, 'class'])` | `new Token(';')`
> -- | -- | --
> "Token kind" | `T_CLASS` | `';'`
> "Token id" | `T_CLASS` | `null`
> "Token content" | `'class'` | `';'`
> "Token prototype" | `[T_CLASS, 'class']` | `';'`
> 


for this part - i like this proposal in general.

i wonder about reusing the `kind` term, that is currently bloated. perhaps worth to fully deprecate it and use different naming. not sure yet.

Reply:
i wouldn't agree with `"Token id": T_CLASS|';'` as it's changing original prototype. i want to extend original type by new properties only, not by messing up native types

Reply:
> let's look if we can clean this up somehow. eg with different naming if they accept more than `kind` only.

When #8925 is merged, I will create a new pr where I only extend the type for the methods of category 1 (including tests for the new type).
After that I will have a closer look to the methods of category 3 and their usages. And will then make some suggestions.

Reply:
i believe for cat 3 we need to either be explicit in naming that we not only allow `kind`, or validate if we actually need to provide `Token` instance as param. (maybe we do, maybe we do not)